cert/selfsigned_test.go

```diff
@@ -185,3 +218,51 @@ func TestTLSDisableAutofill(t *testing.T) {
 		"TLS Certificate was not marked as outdated when it should be",
 	)
 }
+
+// TestTLSConfig tests to ensure we can generate a TLS Config from
+// a tls cert and tls key.
+func TestTLSConfig(t *testing.T) {
+	tempDir := t.TempDir()
+	certPath := filepath.Join(tempDir, "/tls.cert")
+	keyPath := filepath.Join(tempDir, "/tls.key")
+
+	// Generate TLS files with an extra IP and domain.
+	certBytes, keyBytes, err := cert.GenCertPair(
+		"lnd autogenerated cert", certPath, keyPath,
+		[]string{extraIPs[0]}, []string{extraDomains[0]}, false,
+		testTLSCertDuration,
+	)
+	require.NoError(t, err)
+
+	err = cert.WriteCertPair(certPath, keyPath, certBytes, keyBytes)
+	require.NoError(t, err)
+
+	certBytes, err = ioutil.ReadFile(certPath)
+	require.NoError(t, err)
+
+	keyBytes, err = ioutil.ReadFile(keyPath)
+	require.NoError(t, err)
+
+	// Load the certificate.
+	certData, parsedCert, err := cert.LoadCertFromBytes(
+		certBytes, keyBytes,
+	)
+	require.NoError(t, err)
+
+	// Check to make sure the IP and domain are in the cert.
+	var foundIp bool
+	require.Contains(t, parsedCert.DNSNames, extraDomains[0])
+	for _, ip := range parsedCert.IPAddresses {
+		if ip.String() == extraIPs[0] {
```

Comment:
Ah, yes. Some of the golang library's structs aren't comparable directly, unfortunately. Same goes for `time.Time`.